arithmetization/src/main/java/net/consensys/linea/zktracer/types/MemoryRange.java

```diff
@@ -86,8 +86,7 @@ public MemoryRange(
   public MemoryRange(final long contextNumber, final Range range, final MessageFrame frame) {
     this.contextNumber = contextNumber;
     this.range = range;
-    this.rawData =
-        (range.isEmpty()) ? Bytes.EMPTY : frame.shadowReadMemory(0, frame.memoryByteSize());
+    this.rawData = frame.shadowReadMemory(0, frame.memoryByteSize());
```

Comment:
The only fix: as we pass the memory snapshot to the MMU, we need to take a snapshot of the memory even if callData is empty. If not, we don't remember the caller memory when doing a (empty) sha2 for exempel

Reply:
Yep, you right? I modified `rawCallerMemory()` to retrieve the memory from the `returnAtRange`if the `callDataRange` is empty.

Next point, as we almost always have a nin empty callData and a returnAt, wouldn't it be better to have a pointer in each, pointing to the full memory snapshot ? In almost all cases, we'll store double data for nothing.

Reply:
I think it's worth changing down the road. The current solution is clearly sub optimal in terms of memory usage, but it's simple and I don't expect that it will have a performance impact. I would leave things as they are for now.